R/importSNV.TCGA.R

```diff
@@ -0,0 +1,268 @@
+#'   ## importSNV.TCGA(c("aMAFfile.maf","bMAFfile"), "sampleSNV", na.rm = FALSE) ## }
+#' @export
+
+importSNV.TCGA <- function(  fMAF,
+                             rSNV,
+                             na.rm = FALSE,
+                             silent = FALSE,
+                             writeLog = TRUE) {
+
+    myNotes <- character()
+    myCall <- character()
+
+    # ==== PARAMETERS ==========================================================
+
+    # NL <- .PlatformLineBreak()
+    # # Read header and one contents line
+    # tmp <- readLines(fName, n = 2)
+
+    # ==== VALIDATIONS =========================================================
+
+    # General parameter checks
+    cR <- character()
+    cR <- c(cR, .checkArgs(fMAF,         like = rep("FILE_E", length(fMAF)),
+                                                             checkSize = TRUE))
+    cR <- c(cR, .checkArgs(silent,       like = logical(1), checkSize = TRUE))
+    cR <- c(cR, .checkArgs(writeLog,     like = logical(1), checkSize = TRUE))
+    cR <- c(cR, .checkArgs(na.rm,        like = logical(1), checkSize = TRUE))
+
+    if(length(cR) > 0) {
+        stop(cR)
+    }
+
+    # === Validate silent, writeLog, na.rm =================================
+    if (is.numeric(silent)) {
+        stop("silent cannot be numeric : should be logical TRUE/FALSE")
+    } else if (is.null(silent)) {
+        stop("silent cannot be NULL : should be logical TRUE/FALSE")
+    } else if (length(silent) > 1) {
+        stop("silent cannot have more than 1 values : should be logical TRUE/FALSE")
+    }
+
+    if (is.numeric(writeLog)) {
+        stop("writeLog cannot be numeric : should be logical TRUE/FALSE")
+    } else if (is.null(writeLog)) {
+        stop("writeLog cannot be NULL : should be logical TRUE/FALSE")
+    } else if (length(writeLog) > 1) {
+        stop("writeLog cannot have more than 1 values : should be logical TRUE/FALSE")
+    }
+
+    if (is.numeric(na.rm)) {
+        stop("na.rm cannot be numeric : should be logical TRUE/FALSE")
+    } else if (is.null(na.rm)) {
+        stop("na.rm cannot be NULL : should be logical TRUE/FALSE")
+    } else if (length(na.rm) > 1) {
+        stop("na.rm cannot have more than 1 values : should be logical TRUE/FALSE")
+    }
+    # ==== Validata fMAF and rSNV  =======================================
+    if (missing(fMAF)) {
+        stop("missing parameter fMAF with no default: list of MAF file names")
+    } else if (is.null(fMAF)) {
+        stop("parameter fMAF cannot take value \"NULL\": should be a vector/string of MAF file name(s)")
+    } else if (is.numeric(fMAF)) {
+        stop("parameter fMAF cannot be numeric: should be a vector/string of MAF file name(s)")
+    }
+
+    if (missing(rSNV)) {
+        rSNVu <- .makeRandomString()
+
+        write("sym\tchr\tstart\tend\tstrand\tclass\ttype\taRef\ta1\ta2\tTCGA\tUUID",
+              rSNVu, append=FALSE)
+
+        snvValidity <- character()
+    } else if (is.null(rSNV)) {
```

Comment:
Yeah, I thought so too. But from one of your earlier comments about using the second parameter of expect_error() which is a regex. I thought you wanted me to throw my own errors from this function and check for them separately.
Fixed now.